interfaces/dbus/backend.go

```diff
@@ -101,8 +119,19 @@ func (b *Backend) Setup(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 
 	// core/snapd on classic are special
 	if (snapInfo.Type() == snap.TypeOS || snapInfo.Type() == snap.TypeSnapd) && release.OnClassic {
-		if err := setupDbusServiceForUserd(snapInfo); err != nil {
-			logger.Noticef("cannot create host `snap userd` dbus service file: %s", err)
+		// fugly - but we need to make sure that the content of the
+		// "snapd" snap wins
+		//
+		// TODO: this is also racy but the content of the files in core and
+		// snapd is identical cleanup after link-snap and
+		// setup-profiles are unified
+		if !(snapInfo.InstanceName() == "core" && osutil.FileExists(filepath.Join(snapInfo.MountDir(), "../..", "snapd/current"))) {
+			if err := setupDbusServiceForUserd(snapInfo); err != nil {
+				logger.Noticef("cannot create host `snap userd` dbus service file: %s", err)
+			}
+			if err := setupHostDBusConf(snapInfo); err != nil {
```

Comment:
This should probably be be limited to distros that reexec. The mechanism is not that userful in other scenarios.

I looked at the code we have, and it seems there no helper that indicates whether snapd was indeed rexeced. I can try and add one.

Reply:
#8898 introduces a helper to check whether we have reexeced.

Reply:
I've updated the branch to use the new `snapdtool.IsReexecd` helper, and moved all the conditionals to a `shouldCopyConfigFiles` helper function to flatten things a bit and simplify the logic.